packages/ui-components/src/components/Chart/PieChart.js

```diff
@@ -0,0 +1,256 @@
+  const adjustmentAmount = 1.5; // (just derived from eyeballing what looks ok)
+  const adjustmentFactor = Math.min(1, Math.abs(cos * adjustmentAmount));
+
+  // find start of extension line
+  const extensionStartRadius = outerRadius + 10;
+  const sx = cx + extensionStartRadius * cos;
+  const sy = cy + extensionStartRadius * sin;
+
+  // find end of extension line / start of text line
+  const extensionEndRadius = extensionStartRadius + adjustmentFactor * 20;
+  const mx = cx + extensionEndRadius * cos;
+  const my = cy + extensionEndRadius * sin;
+
+  // find end of text line
+  const horizontalTooltipLineLength = 52 + (1 - adjustmentFactor) * 30;
+  const ex = mx + (isLabelOnRight ? 1 : -1) * horizontalTooltipLineLength;
+  const ey = my;
+
+  const textAnchor = isLabelOnRight ? 'start' : 'end';
+  const valueTypeForLabel = labelType || valueType;
+  const metadataForLabel = originalItem[`${name}_metadata`];
+
+  return (
+    <g>
+      <Sector
+        cx={cx}
+        cy={cy}
+        innerRadius={innerRadius}
+        outerRadius={outerRadius}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        fill={fill}
+      />
+      <Sector
+        cx={cx}
+        cy={cy}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        innerRadius={outerRadius + 6}
+        outerRadius={outerRadius + 10}
+        fill={fill}
+      />
+      <path d={`M${sx},${sy} L${mx},${my} L${ex},${ey}`} stroke={fill} fill="none" />
+      <circle cx={ex} cy={ey} r={2} fill={fill} stroke="none" />
+      <text x={ex + (isLabelOnRight ? 1 : -1) * 12} y={ey} textAnchor={textAnchor} fill="#BBB">
+        {name}
+      </text>
+      <text
+        x={ex + (isLabelOnRight ? 1 : -1) * 12}
+        y={ey}
+        dy={18}
+        textAnchor={textAnchor}
+        fill="#999"
+      >
+        {formatDataValue(value, valueTypeForLabel, metadataForLabel)}
+      </text>
+    </g>
+  );
+};
+
+export const PieChart = ({ viewContent, isExporting, isEnlarged, onItemClick }) => {
+  const [activeIndex, setActiveIndex] = useState(-1);
+
+  // Disable tapping charts to reveal legend labels on mobile as they do not fit and are awkwardly cropped
+  const handleMouseEnter = (object, index) => {
+    if (!isMobile()) {
+      setActiveIndex(index);
+    }
+  };
+
+  const handleMouseOut = () => {
+    if (!isMobile()) {
+      setActiveIndex(-1);
+    }
+  };
+
+  const getPresentationOption = (key, option) => {
+    const { presentationOptions } = viewContent;
+    return presentationOptions && presentationOptions[key] && presentationOptions[key][option];
+  };
+
+  const getValidData = () => {
+    const { data, valueType } = viewContent;
+
+    return data
+      .filter(element => element.value > 0)
+      .map(item => {
+        const { name, ...otherKeyValues } = item;
+        // Map names to labels if available
+        let label = getPresentationOption(name, 'label');
```

Comment:
```suggestion
        const label = getPresentationOption(name, 'label') ?? name;
```
Though ?? will treat falsey as valid.